### PR TITLE
GettingStarted Rework

### PR DESCRIPTION
**Pull request by TackAdam**
### Description
Updates to GettingStarted
1. Restructure getting started to be Logs, Metrics, and Traces
2. Remove the dashboard creation
3. Change the workflow order to have Schema / Index Patterns (and create asset button)at the first steps before the user creates there index.
4. Get rid of the query and analyze panel and add them into the workflow.
5. If the integration exist (Currently only for Nginx) give an option to redirect to the integration panel to finish creating it.
6. Break apart the OTEL workflow into logs, metrics, and traces/services.
7. Remove the quickstart workflow and replace it with "Self Managed" and an additional tab for "AWS"(for OTEL logs/metrics/traces)
8. Add default behavior to open up OTEL Logs when landing on the page.
9. Remove workflow for file upload

| Before | After |
| ------ | ----- |
| Before Image 1 | After Image 1 |
| Before Image 2 | After Image 2 |
| Before Image 3 | After Image 3 |
| Before Image 4 | After Image 4 |

Video of new experience
https://github.com/user-attachments/assets/a8cc30d6-70a7-4d18-b8c3-6b7ffada0869

For Nginx
Screenshot 2024-10-10 at 2 46 53 PM
Redirect to 
Screenshot 2024-10-10 at 2 47 12 PM

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: server/routes/getting_started/helper.ts [hunk 1]
```diff
@@ -8,8 +8,12 @@ import path from 'path';
 
 export const assetMapper = (tutorialId: string) => {
```
Comment (mengweieric) [inline server/routes/getting_started/helper.ts:9]: We may want to consider refactor this code piece as it is
- not easy to maintain, the growing number of cases makes these magic strings 
  - proven to typos
  - not easy to be modified when naming and version changes along the way

we want to consider 
- use map or enum for all these file names and move these stuff to constant file.
```typescript
type TutorialId = 'otelLogs' | 'otelMetrics' | 'otelTraces' | 'nginx' | 'java' | 'python' | 'golang';

const assetMap: Record<TutorialId, string> = {
  otelLogs: 'otel-index-patterns-1.0.0-Logs.ndjson',
  otelMetrics: 'otel-index-patterns-1.0.0-Metrics.ndjson',
  otelTraces: 'otel-index-patterns-1.0.0-Traces.ndjson',
  nginx: 'nginx-1.0.0.ndjson',
  java: 'java-tutorial-1.0.0.ndjson',
  python: 'python-tutorial-1.0.0.ndjson',
  golang: 'golang-tutorial-1.0.0.ndjson',
};

export const assetMapper = (tutorialId: TutorialId): string => {
  return assetMap[tutorialId] || '';
};
```
- maybe use concatenations like `${component}-${indexPattens}-${version}-${signal}.ndjson`
- provides a default return?

Also similar question: are these files coming from integration? If so, do we need to maintain these files from within get started module? Can't we just fetch from somewhere?

**Comment by mengweieric** [comment]
Could we also add tests to some of these changes?

File: public/components/getting_started/getting_started_artifacts/otel-services/.DS_Store
[the remark this answers is not in the document]
Comment (TackAdam) [reply public/components/getting_started/getting_started_artifacts/otel-services/.DS_Store:1]: Removed.

File: public/components/getting_started/getting_started_artifacts/nginx/nginx-1.0.0.json [hunk 2]
```diff
@@ -108,88 +108,45 @@
     }
   ],
   "getting-started": {
-    "info": "https://github.com/opensearch-project/opensearch-catalog/blob/main/integrations/observability/nginx/getting-started/Getting-Started.md",
+    "info": {
```
[the remark this answers is not in the document]
Comment (TackAdam) [reply public/components/getting_started/getting_started_artifacts/nginx/nginx-1.0.0.json:111]: This link is not exposed to the user and was previously used to generate information on the dashboard we are not creating anymore. 
We are currently manually making all these json's to provide the steps/workflows but would like to eventually move these to the catalog.

File: public/components/getting_started/components/getting_started.tsx [hunk 3]
```diff
@@ -17,10 +16,8 @@ interface ExtendedHomeProps extends HomeProps {
 
 export const NewGettingStarted = (props: ExtendedHomeProps) => {
   const { chrome, selectedDataSourceId, selectedDataSourceLabel } = props;
-  const [selectedSource, setSelectedSource] = useState('');
+  const [_selectedSource, setSelectedSource] = useState('');
```
[the remark this answers is not in the document]
Comment (TackAdam) [reply public/components/getting_started/components/getting_started.tsx:19]: Was used in the previous format of getting started, have removed it.

File: public/index.scss [hunk 4]
```diff
@@ -13,6 +13,12 @@
 @import 'components/event_analytics/index';
 
 .synopsisIcon {
-    height: 40px;
-    width: 40px;
-  }
\ No newline at end of file
+  height: 40px;
```
[the remark this answers is not in the document]
Comment (TackAdam) [reply public/index.scss:16]: The icon components are shared between integrations and getting-started to keep the image size consistent between the plugins.

File: public/components/getting_started/home.tsx [hunk 5]
```diff
@@ -98,6 +99,9 @@ export const Home = (props: HomeProps) => {
   return (
     <div>
       {dataSourceMenuComponent}
+      <HeaderControlledComponentsWrapper
+        description={'Get started with collecting and monitoring your observability data.'}
```
[the remark this answers is not in the document]
Comment (TackAdam) [reply public/components/getting_started/home.tsx:103]: Added i18n 
```
      <HeaderControlledComponentsWrapper
        description={i18n.translate('observabilityGetStarted.description', {
          defaultMessage: 'Get started with collecting and monitoring your observability data.',
        })}
```
I don't think it should be moved to constant folder as it is a 1 off description and currently all the top nav registration of descriptions are done in file.

File: server/routes/getting_started/helper.ts (hunk 1 above)
Comment (TackAdam) [reply server/routes/getting_started/helper.ts:9]: These are all currently manually created.

Comment (TackAdam) [reply server/routes/getting_started/helper.ts]: Adjusted it to be
```
export const assetMapper = (tutorialId: TutorialId): string => {
  const component = COMPONENT_MAP[tutorialId] || 'default-component';
  const version = VERSION_MAP[tutorialId] || '1.0.0';
  const signal = SIGNAL_MAP[tutorialId] ? `-${SIGNAL_MAP[tutorialId]}` : '';

  return `${component}-${version}${signal}.ndjson`;
};
```
and moved others to constants.

File: public/components/getting_started/getting_started_artifacts/nginx/nginx-1.0.0.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (mengweieric) [reply public/components/getting_started/getting_started_artifacts/nginx/nginx-1.0.0.json:111]: > This link is not exposed to the user and was previously used to generate information on the dashboard we are not creating anymore.

We've spoke about this offline, this is not specifically for this info but overall like do we have to manually create/maintain these json files on the UI side.

> We are currently manually making all these json's to provide the steps/workflows but would like to eventually move these to the catalog.

Could we create issue for this just for tracking purposes?

File: public/index.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (mengweieric) [reply public/index.scss:16]: you could reuse or create shared.scss for general styings that are shared across different module. index file here are for modularity importing scss partials.

[the remark this answers is not in the document]
Comment (mengweieric) [reply public/index.scss:16]: you could reuse or create shared.scss for general styings that are shared across different module. index file here are for modularity importing scss partials.

File: public/components/getting_started/home.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (mengweieric) [reply public/components/getting_started/home.tsx]: thanks, generally we should for better maintainability but not a blocker

**Review by mengweieric — APPROVED** [review]

**Review by paulstn — APPROVED** [review]

**Pull request merged**